src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2404,6 +2412,16 @@ bool primitive_inst::is_valid_fusion() const {
 
             if (gemm_dims[0] != data_dims[0])
                 return false;
+        } else if (_node->is_type<fully_connected>() && _node->get_preferred_impl_type() == impl_types::onednn) {
+            const auto& fc_layout = _impl_params->get_output_layout();
+            const auto& data_layout = outer_dep.first->_impl_params->get_output_layout();
+
+            if ((fc_layout.batch() == 1 || fc_layout.feature() == 1) ||
+                (data_layout.batch() == 1 && data_layout.feature() == 1) ||
+                (fc_layout.count() == data_layout.count())) {
+            } else {
+                return false;
```

Comment:
Nit: 
1. Don't use batch(), feature() methods, address dimensions directly instead.
2. You have empty if() body, so just invert condition to avoid it

Reply:
I updated it as you reviewed. Thank you.